addons/mail/static/src/core/web/thread_service_patch.js

```diff
@@ -230,6 +230,7 @@ patch(ThreadService.prototype, {
         super.unpin(...arguments);
     },
     _openChatWindow(thread, replaceNewMessageChatWindow, { autofocus = true, openMessagingMenuOnClose } = {}) {
+        this.store.env.services["bus_service"].addChannel(thread.busChannel);
```

Comment:
This feels necessary because you by-pass `is_pinned` from server. Calling `joinChat`/`joinChannel` rather than `fetchChannel` would fetch and also properly mark channel as pinned on UI, thus bus notification changes are received as they should.

Reply:
I think it's fine, for livechat history (as non-member) we relied on `isLocallyPinned` to do a manual `addChannel`, but it this case we just want to open the chat window, not actually pin the channel.

Reply:
We might want to only do this for channel however, not for other threads.

Reply:
So this should be moved as a patch on `discuss` subfolder with proper check on thread model (= channel).

Reply:
And I don't think it should be `web` specifically as chat window are now in `common`? But as the base method is already in web I guess we can't do much in this version, just see to clean in master.

However we might also want to ensure we don't call this every time we open a chat window, but only if it is necessary? (= channel not pinned?) as it would force resubscribe the bus every time otherwise.

Reply:
Made the changes as you suggested